app/Models/Port.php

```diff
@@ -12,7 +13,40 @@ class Port extends DeviceRelatedModel
     public $timestamps = false;
     protected $primaryKey = 'port_id';
 
-    // ---- Helper Functions ----
+    /**
+     * Initialize this class
+     */
+    public static function boot()
+    {
+        parent::boot();
+
+        static::deleting(function (Port $port) {
+            // delete related data
+            $port->adsl()->delete();
+            $port->fdbEntries()->delete();
+            $port->ipv4()->delete();
+            $port->ipv6()->delete();
+            $port->macAccounting()->delete();
+            $port->macs()->delete();
+            $port->nac()->delete();
+            $port->ospfNeighbors()->delete();
+            $port->ospfPorts()->delete();
+            $port->pseudowires()->delete();
+            $port->statistics()->delete();
+            $port->stp()->delete();
+            $port->vlans()->delete();
+
+            // dont have relationships yet
+            DB::table('juniAtmVp')->where('port_id', $port->port_id)->delete();
+            DB::table('ports_perms')->where('port_id', $port->port_id)->delete();
+            DB::table('links')->where('local_port_id', $port->port_id)->orWhere('remote_port_id', $port->port_id)->delete();
+            DB::table('ports_stack')->where('port_id_low', $port->port_id)->orWhere('port_id_high', $port->port_id)->delete();
+
+            \Rrd::purge($port->device->hostname, \Rrd::portName($port->port_id)); // purge all port rrd files
```

Comment:
Tests fail because `$port->device` is undefined, would be cool if we could remove port RRD without involving device

Reply:
Can't remove the rrd without knowing the hostname.  We do use eager loading, so it is only one extra query (per chunk).

Reply:
->device was undefined or null?  because I could see null if the device got deleted orphaning the ports.  Not sure how it would be undefined though.